src/cunumeric/scan/scan_local_template.inl

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2021-2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/scan/scan_local_util.h"
+#include "cunumeric/pitches.h"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <VariantKind KIND, ScanCode OP_CODE, LegateTypeCode CODE, int DIM>
+struct ScanLocalImplBody;
+
+template <VariantKind KIND, ScanCode OP_CODE, LegateTypeCode CODE, int DIM>
+struct ScanLocalNanImplBody;
+
+template <VariantKind KIND, ScanCode OP_CODE, bool NAN0>
+struct ScanLocalImpl {
+  // Case where NANs are transformed
+  template <LegateTypeCode CODE,
+            int DIM,
+            std::enable_if_t<NAN0 && legate::is_floating_point<CODE>::value>* = nullptr>
```

Comment:
I did not enable complex type check for the nan versions of code as this would cause a lot of headaches with std::isnan and making sure cunumeric behaves consistent with numpy, that's why in python layer I've set these not implemented for this PR.

Reply:
6bae8cb705df10a6a4688a993c16f22bb83c5da1